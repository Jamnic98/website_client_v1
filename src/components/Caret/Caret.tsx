import classNames from 'classnames'

import styles from './caret.module.css'

export enum CaretDirection {
  UP = 'up',
  DOWN = 'down',
  LEFT = 'left',
  RIGHT = 'right',
}

export interface CaretProps {
  direction?: CaretDirection
  isFlipped?: boolean
}

export const Caret: React.FC<CaretProps> = ({
  direction = CaretDirection.UP,
  isFlipped = false,
}: CaretProps) => (
  <div
    className={classNames(styles.caret, styles[direction], {
      [styles.flip]: isFlipped,
    })}
  >
    ^
  </div>
)
