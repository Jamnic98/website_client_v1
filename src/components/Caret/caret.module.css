.caret {
  display: inline-block;
  position: relative;
  transition: transform 0.2s ease;
  font-family:
    system-ui,
    -apple-system,
    BlinkMacSystemFont,
    'Segoe UI',
    Roboto,
    Oxygen,
    Ubuntu,
    Cantarell,
    'Open Sans',
    'Helvetica Neue',
    sans-serif;
  width: none;
  height: none;
  line-height: 0;
}

.up {
  transform: rotate(0deg);
}

.down {
  transform: rotate(180deg);
}

.left {
  transform: rotate(-90deg);
}

.right {
  transform: rotate(90deg);
}

.up.flip {
  transform: rotate(180deg);
}

.down.flip {
  transform: rotate(0deg);
}

.left.flip {
  transform: rotate(90deg);
}

.right.flip {
  transform: rotate(270deg);
}
