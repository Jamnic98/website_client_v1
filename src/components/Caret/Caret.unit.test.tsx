import { render, screen } from '@testing-library/react'

import { Caret, type CaretProps } from './Caret'

const testData = { isFlipped: false }

const getTestCaret = (props?: CaretProps) => {
  const finalProps = {
    isFlipped: testData.isFlipped,
    ...props,
  }
  return <Caret {...finalProps} />
}

const drawTestCaret = (props?: CaretProps) => {
  return render(getTestCaret(props))
}

describe('test Caret', () => {
  it('displays correctly', () => {
    drawTestCaret()
    const caret = screen.getByText('^')
    expect(caret).toBeInTheDocument()
  })

  it('rotates 180 degrees when isFlipped = true', () => {
    const { rerender } = drawTestCaret()
    const caret = screen.getByText('^')
    rerender(getTestCaret({ isFlipped: true }))
    expect(caret).toHaveClass('flip')
  })
})
