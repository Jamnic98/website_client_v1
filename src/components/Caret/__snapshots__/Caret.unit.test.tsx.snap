// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Caret renders Caret unchanged 1`] = `
<div
  className="caret up"
>
  ^
</div>
`;
